File: Earliest Finish Time for Land and Water Rides I/Earliest Finish Time for Land and Water Rides I.java
// Solution for Earliest Finish Time for Land and Water Rides I in JAVA

class Solution {
    public int earliestFinishTime(int[] landStartTime, int[] landDuration,
                                  int[] waterStartTime, int[] waterDuration) {
        int ans = Integer.MAX_VALUE;

        for (int i = 0; i < landStartTime.length; i++) {
            int landEnd = landStartTime[i] + landDuration[i];

            for (int j = 0; j < waterStartTime.length; j++) {
                int waterEnd = waterStartTime[j] + waterDuration[j];

                int landWater = Math.max(landEnd, waterStartTime[j]) + waterDuration[j];
                int waterLand = Math.max(waterEnd, landStartTime[i]) + landDuration[i];

                ans = Math.min(ans, Math.min(landWater, waterLand));
            }
        }
        return ans;
    }
}

File: Earliest Finish Time for Land and Water Rides I/Earliest Finish Time for Land and Water Rides I.py
# Solution for Earliest Finish Time for Land and Water Rides I in PY

class Solution:
    def earliestFinishTime(self, landStartTime, landDuration,
                           waterStartTime, waterDuration):
        ans = float('inf')

        for i in range(len(landStartTime)):
            land_end = landStartTime[i] + landDuration[i]

            for j in range(len(waterStartTime)):
                water_end = waterStartTime[j] + waterDuration[j]

                ans = min(
                    ans,
                    max(land_end, waterStartTime[j]) + waterDuration[j],
                    max(water_end, landStartTime[i]) + landDuration[i]
                )

        return ans
